src/pcre2_compile.c

```diff
@@ -10549,7 +10552,15 @@ if (has_lookbehind)
     }
   memset(cb.groupinfo, 0, (2 * cb.bracount + 1) * sizeof(uint32_t));
   errorcode = check_lookbehinds(cb.parsed_pattern, NULL, NULL, &cb, &loopcount);
-  if (errorcode != 0) goto HAD_CB_ERROR;
+  if (errorcode != 0)
+    {
+#ifdef PCRE2_DEBUG
+    /* BUG: check_lookbehinds() is missing code for a valid META */
+    if (errorcode == ERR70) cb.erroroffset = PCRE2_UNSET;
```

Comment:
it might be better to directly abort() here to help find this bug

Reply:
Can this commit be applied on top of rel 10.43 as patch? (or should I update the whole thing).

Reply:
the critical part of the "fix" applies cleanly all the way to 10.30:

```
diff -ur pcre2-10.30/src/pcre2_compile.c pcre2-10.30.p/src/pcre2_compile.c
--- pcre2-10.30/src/pcre2_compile.c	2017-08-12 09:18:55
+++ pcre2-10.30.p/src/pcre2_compile.c	2024-06-14 15:49:14
@@ -8858,6 +8858,11 @@
   switch (META_CODE(*pptr))
     {
     default:
+
+    /* The following erroroffset is a bogus but safe value.
+    This branch should be avoided by providing a proper
+    implementation for all supported cases below. */
+    cb->erroroffset = 0;
     return ERR70;  /* Unrecognized meta code */
 
     case META_ESCAPE:
```

the full patch applies cleanly to 10.43 or any later versions.